### PR TITLE
Doc for envsubst deployer

### PR DESCRIPTION
**Pull request by gibbleyg**

### CODE REVIEW

File: docs/building-deployer-envsubst.md [hunk 1]
```diff
@@ -1,3 +1,226 @@
...
+The `manifests.yaml.template` file must be modified to reference `nginxImage`
+
+```yaml
+# Under spec.template
+    spec:
+      containers:
+      - name: my-nginx
+        image: $nginxImage # Changed from "nginx"
+```
+
```
Comment (gibbleyg) [inline docs/building-deployer-envsubst.md:219]: @huyhg I omitted the following lines from the helm deployer guide. Not sure if this still applies. If it does where are "default values" specified for images using `envsubst`?

https://github.com/GoogleCloudPlatform/marketplace-k8s-app-tools/blame/master/docs/building-deployer-helm.md#L406-L414

File: docs/building-deployer-envsubst.md [hunk 2]
```diff
@@ -1,3 +1,226 @@
...
+You __should not__ use images that are not under your control.
+If your application uses some commonly available image like
+`busybox`, make a copy of that image and ensure that it is free
+of any vulnerability or malicious code.
+
+Note that end users never see or need access to these default
+image names.
```
Comment (gibbleyg) [inline docs/building-deployer-envsubst.md:226]: @huyhg , I could copy Steps 3-5 in the deployer-helm guide but those steps do not appear to be specific to `envsubst`. Any issue with me just linking those steps at the end of the `envsubst` guide?

File: docs/building-deployer-envsubst.md [hunk 3]
```diff
@@ -1,3 +1,226 @@
...
+properties:
+  name:
+    type: string
+    x-google-marketplace:
+      type: NAME
+  namespace:
+    type: string
+    x-google-marketplace:
+      type: NAMESPACE
+  replicas:
```
[the remark this answers is not in the document]
Comment (gibbleyg) [reply docs/building-deployer-envsubst.md:51]: Fixed

File: docs/building-deployer-envsubst.md [hunk 4]
```diff
@@ -50,13 +50,22 @@ properties:
...
+    title: Nginx Replica Count
+    description: The number of nginx replicas to deploy
 
 required:
 - name
 - namespace
 - replicas
 ```
 
+The `name` and `namespace` properties are required for all applications, and
```
Comment (huyhg) [inline docs/building-deployer-envsubst.md:62]: nit: the specific names, `name` and `namespace`, can be arbitrary.

File: docs/building-deployer-envsubst.md (hunk 2 above)
Comment (huyhg) [reply docs/building-deployer-envsubst.md:253]: SGTM.

**Comment by gibbleyg** [comment]
@googlebot rescan

**Pull request merged**